src/ui/application-window.vala

```diff
@@ -136,6 +147,21 @@ namespace Peek.Ui {
       settings.bind ("persist-save-folder",
         this, "save_folder",
         SettingsBindFlags.DEFAULT);
+        
+      //Grab current format and udate the recorder button label to reflect the user's choice  
+      format=settings.get_string("recording-output-format");
+        if (format == "gif") {
```

Comment:
See my comment below for adding a `select_format()` method. This could be reused here to avoid duplication of generating the label texts.

In addition I have constants defined for the different formats in [defaults.vala](https://github.com/phw/peek/blob/master/src/defaults.vala#L18)